app/components/network_tables.py

```diff
@@ -0,0 +1,140 @@
+    NTCORE = False
+
+
+class NumpyEncoder(json.JSONEncoder):
+    def default(self, o):
+        if isinstance(o, np.ndarray):
+            return o.tolist()
+        elif isinstance(o, (np.integer,)):
+            return int(o)
+        elif isinstance(o, (np.floating,)):
+            return float(o)
+        elif isinstance(o, (np.bool_)):
+            return bool(o)
+        return super().default(o)
+
+
+@singleton
+class NetworkTablesPublisher:
+    def __init__(self):
+        if not NTCORE:
+            return
+        config = ConfigManager().get()
+
+        table_name = config.network_tables.table
+        server = config.network_tables.server
+
+        logger.info(
+            f"Initializing NetworkTables connection to {server} with table {table_name}",
+            operation="init",
+            status="starting",
+        )
+        try:
+            self._initialize_connection(table_name, server)
+            self.connected = True
+            self.publish_count = 0
+            self.error_count = 0
+
+            logger.info(
+                "NetworkTables initialized successfully",
+                operation="init",
+                status="success",
+            )
+        except Exception as e:
+            self.connected = False
+            logger.exception(
+                f"Failed to initialize NetworkTables: {e}", operation="init"
+            )
+            raise
+
+    @retry_with_backoff(max_attempts=3, initial_delay=1.0)
+    def _initialize_connection(self, table_name, server):
+        """Initialize NetworkTables connection with retry logic."""
+        if not NTCORE:
+            return
+        logger.debug("Creating NetworkTables instance", operation="init_connection")
+
+        self.inst = ntcore.NetworkTableInstance.getDefault()
+        self.inst.startClient4(table_name)
+        self.inst.setServer(server)
+
+        logger.debug(
+            f"NetworkTables client started for server {server}",
+            operation="init_connection",
+        )
+
+        self.table = self.inst.getTable(table_name)
+        logger.debug(
+            f"NetworkTables table '{table_name}' acquired", operation="init_connection"
+        )
+
+        # Struct array publisher instead of single struct
+        self.pose_pub = self.table.getStructArrayTopic("poses", Pose3d).publish()
+        logger.debug("Pose publisher created", operation="init_connection")
+
+    def publish_detections(self, detections: list[Detection]):
+        """Publish detection results to NetworkTables with error handling."""
+        if not NTCORE:
+            return
+        try:
+            if not detections:
+                self.clear()
+                return
+
+            poses = []
+            for i, det in enumerate(detections):
+                try:
+                    point = det.point
+                    pose = Pose3d(Translation3d(point.x, point.y, -point.z), Rotation3d(0.0, 0.0, 0.0))
```

Comment:
```suggestion
                    pose = Pose3d(Translation3d(point.x, point.y, -point.z), Rotation3d())
```
Or can the rotation be removed entirely?

Reply:
e76798c566696f9ae8cd7b52918a53282982b0c2
[Seems like it can't be removed entirely](https://robotpy.readthedocs.io/projects/wpimath/en/latest/wpimath.geometry/Pose3d.html#wpimath.geometry.Pose3d)